src/screens/AddVocab/AddVocab.jsx

```diff
@@ -1,198 +1,315 @@
 
   return (
-    <div className="add-vocab-form">
-      <h1 className="heading">{t("screens.addVocab.title")}</h1>
+    <>
+      <div className="add-vocab-form">
+        <h1 className="heading">{t("screens.addVocab.title")}</h1>
 
-      <div className="dropdowns">
-        <div className="select-wrapper">
-          <Select
+        <div className="dropdowns">
+          <div className="select-wrapper">
+            <Select
+              required
+              creatable
+              createText={"Create new Package"}
+              onCreate={openPackageModal}
+              tabIndex={1}
+              label={t("global.package")}
+              options={packageItems}
+              onChange={(v) => {
+                setSelectedPackage(v);
+              }}
+              value={selectedPackage}
+              noOptionsMessage={t("screens.addVocab.noPackagesMessage")}
+            />
+          </div>
+          <div className="select-wrapper">
+            <Select
+              required
+              creatable
+              createText={"Create new Group"}
+              onCreate={openGroupModal}
+              disabled={!selectedPackage}
+              tabIndex={1}
+              label={t("global.group")}
+              options={groupsItems}
+              onChange={(v) => {
+                setSelectedGroup(v);
+              }}
+              value={selectedGroup}
+              noOptionsMessage={t("screens.addVocab.noGroupsMessage")}
+            />
+          </div>
+        </div>
+        <div className="input-fields">
+          <TextInput
             required
             tabIndex={1}
-            label={t("global.package")}
-            options={languagePackageDropdownItems}
+            placeholder={t("global.foreignWord")}
             onChange={(value) => {
-              // setPackageName(value);
-              // getGroups(e.target.value);
-              console.log("changed: ", value);
+              setForeignWord(value);
             }}
-            value={packageName}
-            noOptionsMessage={t("screens.addVocab.noPackagesMessage")}
+            value={foreignWord}
           />
-        </div>
-        <div className="select-wrapper">
-          <Select
+          <ArrayTextInput
             required
+            max={maxTranslations}
+            data={translations}
+            placeholder={t("global.translation")}
+            onChange={setTranslations}
+            addText={t("screens.addVocab.addTranslation")}
+          />
+          <TextInput
             tabIndex={1}
-            label={t("global.group")}
-            options={groupDropdownItems}
+            placeholder={t("global.description")}
             onChange={(value) => {
-              setGroupName(value);
+              setDescription(value);
             }}
-            value={groupName}
-            noOptionsMessage={t("screens.addVocab.noGroupsMessage")}
+            value={description}
+          />
+          <Switch
```

Comment:
could you try to make the switch a bit smaller?

Reply:
done